path_service_principal_test.go

```diff
@@ -1019,7 +1020,7 @@ func TestCredentialInteg_msgraph(t *testing.T) {
 		// Verify that SP get is an error after delete. Expected there
 		// to be a delay and that this step would take some time/retries,
 		// but that seems not to be the case.
-		assertServicePrincipalDoesNotExist(t, provider.spClient, spObjID)
+		assertServicePrincipalExists(t, provider.spClient, spObjID, false)
```

Comment:
this is failing with
```
    path_service_principal_test.go:1023:
        unexpected error: Resource '1252c6cf-ca96-4de1-ad73-c7b60cea200c' does not exist or one of its queried reference-property objects are not present.
--- FAIL: TestCredentialInteg_msgraph (0.00s)
    --- FAIL: TestCredentialInteg_msgraph/service_principals (27.03s)
    
```

Should we be asserting that it Does Not Exist?

Reply:
Oh, I see we are passing false for the last parameter. I missed that. Should we separate them into separate functions like they were before? Either way, I think we will want to not fail when we get an error in this helper if we are expecting the resource to not exist.

Reply:
Fixed in latest commit. I tested things locally and the test passes now after ensuring the correct behavior on resource non-existence. Thanks!